library/data_structures/seg_tree_uncommon/find_last.hpp

```diff
@@ -0,0 +1,34 @@
+/** @file */
+#pragma once
+/**
+ * @param le,ri defines range [le, ri)
+ * @param f defines a function that returns true if the subtree contains an
+ * element that satisfies the condition
+ * @returns the index of the last element in the range that satisfies the
+ * condition described in `f`, if no such element exists then (le - 1) is
+ * returned
+ * @time O(log(n))
+ * @space O(log(n)) for recursion stack; no new nodes are allocated
+ */
+template <class F>
+int find_last(int le, int ri, const F& f) {
```

Comment:
do we want `const F&` or just `F`, cuz everywhere else it's the latter

https://github.com/programming-team-code/programming_team_code/blob/main/data_structures/rmq.hpp#L18

Reply:
so compare these

not const ref: https://codeforces.com/contest/1705/submission/250982914
with const ref: https://codeforces.com/contest/1705/submission/250940123

you'll notice with const ref is a bit faster so I would vote to keep it (it's better for the recursion stack I think)

in the example you linked it's not recursive so I think it matters less

Reply:
ah I see